[WIKI] study Parameters - Identifiers (#4300)
aces/Loris: docs/wiki/01 - STUDY PARAMETERS SETUP/01 - Study Variables/01 - Identifiers.md

# Identifiers

## Overview
By default LORIS provides 2 different identifiers for each participant or subject: 

- **CandID** also known as the **DCCID** : is a unique randomized 6-digit numeric ID (e.g. '436792').  It is completely anonymous, and is assigned automatically by Loris upon participant registration. 

> - **PSCID** (Project Study Centre ID) is configurable, can be manually entered when registering a participant, and may contain the site abbreviation, followed by sequential or randomized digits (e.g. 'MTL0006')

Reviewer: ```suggestion
- **PSCID** (Project Study Centre ID) is configurable, can be manually entered when registering a participant, and may contain the site abbreviation, followed by sequential or randomized characters (e.g. 'MTL0006')
```

The ID part can be letters or numbers so "characters" is better here.

Reviewer: Yep. Sequential alpha goes `AAA`, `AAB`, `AAC` just like the numbers.

Author: sold

### SQL
_not yet available_

### Adding identifiers from the config.xml file

Both the format and the generation of PSCIDs can be configured by the project in the `config.xml` file of the `%LORIS_ROOT%/project/` directory.

PSCIDs can be created for new subjects in one of 3 ways: *sequentially generated*, *manually entered*, or *randomly generated*.

1. ***sequential*** generates PSCIDs sequentially for each new participant registered. **(default)**

 ```xml
 <PSCID>
     <generation>sequential</generation> 
     <structure>
         <seq type="siteAbbrev"/>
         <seq type="numeric" minLength="4"/>
     </structure>
 </PSCID>
 ```

2. ***manual*** asks the user to enter the PSCID when registering a new participant.

 ```xml
 <PSCID> 
     <generation>user</generation> 
     <structure>
         <seq type="alphanumeric" minLength="4"/> <!-- Ex: AAA1-->
     </structure>
 </PSCID>
 ```

3. ***random*** generates PSCIDs with random numerical for each new participant registered.

 ```xml
 <PSCID>
     <generation>random</generation> 
     <structure>
         <seq type="siteAbbrev"/>
         <seq type="numeric" minLength="4"/>
     </structure>
 </PSCID>
 ```
 

>  Options for the `type` element of the `<seq>` tag are:

Reviewer: You can also specify a `min` and `max` value that represent the upper and lower bounds of an ID. By default it is the maximum value of the alphabet and the length e.g. "ZZZ" for a 3 character alphabetical ID.

Reviewer: @ridz1208 I ended up taking over that and did a bigger rewrite on ID generation https://github.com/aces/Loris/pull/4241

Author: then I'll take a look at your PR and see what adjustments need to be done.

>   - `siteAbbrev`: will draw from Alias field, in psc table

Reviewer: ```suggestion
  - `siteAbbrev`: will draw from Alias field, in the `psc` table
```

  - `numeric`: defines an integer, must be associated with a `minLength` element. 

>   - `alphanumeric`: defines a string, must be associated with a `minLength` element.

Reviewer: There's also an `alpha` type which is purely letters.

# Identifiers

## Overview
By default LORIS provides 2 different identifiers for each participant or subject: 

> - **CandID** also known as the **DCCID** : is a unique randomized 6-digit numeric ID (e.g. '436792').  It is completely anonymous, and is assigned automatically by Loris upon participant registration. 

Reviewer 2: maybe mention that it is not configurable for now or if people want to have their DCCID longer than 6 digits mention how they can do that?

> - **CandID** also known as the **DCCID** : is a unique randomized 6-digit numeric ID (e.g. '436792').  It is completely anonymous, and is assigned automatically by Loris upon participant registration. 

Reviewer: Yeah it's not configurable and it would be good to clarify that.

The format and the generation of the **PSCID** must be configured by an admin at the setup stage.

## Configuration

### Front End
_not yet available_

### SQL
_not yet available_

### Adding identifiers from the config.xml file

Both the format and the generation of PSCIDs can be configured by the project in the `config.xml` file of the `%LORIS_ROOT%/project/` directory.

PSCIDs can be created for new subjects in one of 3 ways: *sequentially generated*, *manually entered*, or *randomly generated*.

1. ***sequential*** generates PSCIDs sequentially for each new participant registered. **(default)**

 ```xml
 <PSCID>
     <generation>sequential</generation> 
     <structure>
         <seq type="siteAbbrev"/>
         <seq type="numeric" minLength="4"/>
     </structure>
 </PSCID>
 ```

2. ***manual*** asks the user to enter the PSCID when registering a new participant.

 ```xml
 <PSCID> 
     <generation>user</generation> 
     <structure>
         <seq type="alphanumeric" minLength="4"/> <!-- Ex: AAA1-->
     </structure>
 </PSCID>
 ```

> 3. ***random*** generates PSCIDs with random numerical for each new participant registered.

Reviewer 2: ... with **a** numerical **value** for each ...

 ```xml
 <PSCID>
     <generation>random</generation> 
     <structure>

>          <seq type="static">PROJ</seq>

Author: @johnsaigle can you confirm that this is how the static type would be defined ?

Reviewer: Yep, that's it!

         <seq type="siteAbbrev"/>
         <seq type="numeric" minLength="4"/>
     </structure>
 </PSCID>
 ```
 
 Options for the `type` element of the `<seq>` tag are:
  - `siteAbbrev`: will draw from Alias field, in the `psc` table

>   - `static`: will draw from the value defined in the `config.xml` file.

Reviewer: ```suggestion
  - `static`: A string value that will be used as a fixed prefix. Value defined in the `config.xml` file.
```

Author: if we rephrase that one we should rephrase all the others to be similar in structure

Reviewer: Yeah I think that's a good idea. I think it's more more important to explain the meaning of each value rather than specify where it's set.

Reviewer: Yeah I think that's a good idea. I think it's more more important to explain the meaning of each value rather than specify where it's set. There's a line earlier in the file that says that these values are all found in config.xml

Author: maybe I'll add the couple missing to the config.xml as an example and I'll try to rephrase these
